basins.py

```diff
@@ -183,22 +202,65 @@
     ]
 )
 
+
+
 basins = regionmask.Regions(
-    [SO, ATL, PAC1, PAC2, ARC, IND, MED, BLK, HUD, BAL, RED, PER]
+    [SO, ATL, PAC1, PAC2, ARC, IND, MED, BLK, HUD, BAL, RED, PER],
+    [ind_SO, ind_ATL, ind_tmp, ind_PAC, ind_ARC, ind_IND, ind_MED,
+     ind_BLK, ind_HUD, ind_BAL, ind_RED, ind_PER]
 )
 
-# define 1x1 grid
-lon = np.arange(0.5, 360.5)
-lat = np.arange(89.5, -90.5, -1)
-
-# make plot
-fig = plt.figure(figsize=(25, 10))
-ax = plt.subplot(111, projection=ccrs.PlateCarree())
-mask = basins.mask(lon, lat)
-# pcolormesh does not handle NaNs, requires masked array
-mask_ma = np.ma.masked_invalid(mask)
-h = ax.pcolormesh(lon, lat, mask_ma, transform=ccrs.PlateCarree(), cmap="jet",)
-basins.plot_regions(ax=ax, add_label=False)
-ax.coastlines()
-
-plt.savefig("basins.png")
+
+
+def generate_basin_codes(grid, lon="geolon", lat="geolat", mask="wet"):
+
+    lon_raw = grid[lon]
+    lon = xr.where(lon_raw < -180, lon_raw+360., lon_raw)
+    lat = grid[lat]
+
+    codes = basins.mask(lon, lat)
+    codes = xr.where(codes == ind_tmp, ind_PAC, codes) # join PAC1 and PAC2
+
+    if mask in grid:
+        masked = grid[mask].values
+        codes = xr.where(masked == 0, 0, codes)
+    else:
+        codes.fillna(0)
+
+    codes['lon'] = lon_raw
+    return codes
+
+
+
+if __name__ == '__main__':
+
+    # define 1x1 grid
```

Comment:
@raphaeldussin - should we add a tripolar example to this?

Reply:
I'd like to add a parser to the main and write tests for one degree (generated on the fly) and OM4 (that we can download
from the GFDL ftp)